core/object/undo_redo.cpp

```diff
@@ -316,6 +316,14 @@ void UndoRedo::commit_action(bool p_execute) {
 	_redo(p_execute); // perform action
 	committing--;
 
+	if (max_steps > 0 && actions.size() > max_steps) {
```

Comment:
```suggestion
	if (max_steps > 0) {
```
The second check is already handled by `while`.